src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -1420,7 +1422,7 @@ private class OffsetAndRule {
             return localConverted;           
         }
 
-
+#if FEATURE_WIN32_REGISTRY
```

Comment:
Is this the right condition?  The code you moved above that's effively just !PLATFORM_UNIX is calling the function in this section.